### PR TITLE
[REFACTOR] proper create_invoice responses for descriptions

### PR DESCRIPTION
**Pull request by dni**
return a exception per error it is description hash or unhashed_description

### CODE REVIEW

File: lnbits/core/views/api.py [hunk 1]
```diff
@@ -196,25 +196,27 @@ class CreateInvoiceData(BaseModel):
 
 
 async def api_payments_create_invoice(data: CreateInvoiceData, wallet: Wallet):
+    description_hash = None
```
Comment (motorina0) [inline lnbits/core/views/api.py:199]: - previously the default values were `b""` and not `None`

Comment (callebtc) [reply lnbits/core/views/api.py:199]: Agree with @motorina0, let's not switcherydoo the types like Python programmers

Comment (dni) [reply lnbits/core/views/api.py]: done

File: lnbits/core/views/api.py [hunk 2]
```diff
@@ -187,38 +187,37 @@ async def api_payments_paginated(
...
+                    status_code=HTTPStatus.BAD_REQUEST,
+                    detail="'description_hash' must be a valid hex strings",
+                )
+        if data.unhashed_description:
+            try:
+                unhashed_description = bytes.fromhex(data.unhashed_description)
+            except ValueError:
+                raise HTTPException(
+                    status_code=HTTPStatus.BAD_REQUEST,
+                    detail="'unhashed_description' must be a valid hex strings",
```
Comment (michael1011) [inline lnbits/core/views/api.py:209]:
```suggestion
                    detail="'unhashed_description' must be a valid hex string",
```

**Review by arcbtc — APPROVED** [review]

**Review by jackstar12 — APPROVED** [review]

**Pull request merged**